BlueDBOnDisk/src/main/java/io/bluedb/disk/recovery/TimeStampedFile.java

```diff
@@ -0,0 +1,36 @@
+package io.bluedb.disk.recovery;
+
+import java.io.File;
+
+public class TimeStampedFile implements Comparable<TimeStampedFile> {
+	private final Long timestamp;
+	private final File file;
+
+	public TimeStampedFile (File file) {
+		this.file = file;
+		this.timestamp = extractTimestamp(file);
+	}
+
+	public File getFile() {
+		return file;
+	}
+
+	public Long getTimestamp() {
+		return timestamp;
+	}
+
+	@Override
+	public int compareTo(TimeStampedFile o) {
+		return timestamp.compareTo(o.timestamp);
+	}
+
+	public static Long extractTimestamp(File file) {
+		try {
+			String fileName = file.getName();
+			String longString = fileName.split("[.]")[0];
+			return Long.valueOf(longString);
+		} catch (Throwable t) {
+			return null;

```

Comment:
I still don't love using "null" as a Long value. It feels risky. I'm okay to merge it, but I do think it'd be worth considering using a special case value like -1. But I don't care enough to hold up the merge.